cvat/apps/dataset_manager/bindings.py

```diff
@@ -1892,7 +1893,7 @@ def load_dataset_data(project_annotation, dataset: dm.Dataset, project_data):
     else:
         for label in dataset.categories()[dm.AnnotationType.label].items:
             if not project_annotation.db_project.label_set.filter(name=label.name).exists():
-                if label.name == 'background':
+                if label.name == "background" and len(deepcopy(dataset).filter("/item[annotation/label=background]")) == 0:
```

Comment:
Copying a multigigabyte datasets in a loop doesn't seem right. Also, the background class annotations are almost guaranteed to exist in an average dataset. Probably, we can just skip these annotations if the class is not declared in the task, because it is mostly going to affect holes in objects. On exporting back such annotations, the background class and the holes will be added automatically. There [is another problem](https://github.com/opencv/cvat/issues/5137) with masks containing holes, but it must be resolved on its own.

Reply:
You're right. Added transformation to remove annotations with `background` class.